webclient/http1/src/main/java/io/helidon/webclient/http1/Http1ClientRequestImpl.java

```diff
@@ -203,6 +203,22 @@ Http1ClientImpl http1Client() {
         return http1Client;
     }
 
+    /**
+     * Check upgrade protocols. Protocol names are case insensitive.
+     *
+     * @param requestUpgrade request upgrade header
+     * @param responseUpgrade response upgrade header
+     * @return check if protocol upgrade can proceed
+     */
+    static boolean isUpgradeSuccessful(Header requestUpgrade, Header responseUpgrade) {
+        for (String protocol : requestUpgrade.allValues()) {
+            if (protocol.equalsIgnoreCase(responseUpgrade.get())) {
```

Comment:
Maybe store the value of `responseUpgrade.get()` in a local variable outside of the loop?